docs/architecture.md

```diff
@@ -0,0 +1,21 @@
+# Profiler Architecture
+
+This document provides an overall look at the architecture of the profiler, and how all the various pieces fit together.
+
+## Gecko Profiler
+
+![Gecko Profiler diagram](images/gecko-profiler.svg)
+
+## Browser
+
+![Browser diagram](images/browser.svg)
+
+## DevTools Recording Panel
+
+Please note, this is the diagram for the new recording panel in DevTools that is off by default. This is the path forward for integrating perf.html into DevTools. It can be enabled using `devtools.performance.new-panel-enabled`. There is no architecture doc for the current Firefox DevTools panel.
+
+![DevTools Recording Panel diagram](images/devtools-recording-panel.svg)
```

Comment:
Like others, I'm confused with the dotted line. I don't understand what it means. Do we use this link? Or do you want to outline this link does exist within the devtools but that we don't use it? Maybe some actual text explaining this would be better.